arrow/flight/gen.go

```diff
@@ -16,5 +16,5 @@
 
 package flight
 
-//go:generate protoc --experimental_allow_proto3_optional -I../../../format --go_out=./gen/flight --go-grpc_out=./gen/flight --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative Flight.proto
-//go:generate protoc --experimental_allow_proto3_optional -I../../../format --go_out=./gen/flight --go-grpc_out=./gen/flight --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative FlightSql.proto
+//go:generate protoc --experimental_allow_proto3_optional -I${APACHE_ARROW_FORMAT_DIR} --go_out=./gen/flight --go-grpc_out=./gen/flight --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative Flight.proto
+//go:generate protoc --experimental_allow_proto3_optional -I${APACHE_ARROW_FORMAT_DIR} --go_out=./gen/flight --go-grpc_out=./gen/flight --go_opt=paths=source_relative --go-grpc_opt=paths=source_relative FlightSql.proto
```

Comment:
Can we move these `protoc ...` command lines to `dev/sync_flight.sh` and remove this file?

Reply:
the standard way of handling things like this would be to utilize `go generate` to run the commands rather than a separate sync script (ideally). So I'd personally prefer keeping this file and the `go:generate` lines.

Reply:
that said, it might be better to utilize `buf` instead see the links below on the `substrait-go` repo for how I did it there to reference and generate the protobuf from the .proto files in the `substrait-io/substrait` repo

https://github.com/substrait-io/substrait-go/blob/main/doc.go#L9
https://github.com/substrait-io/substrait-go/blob/main/buf.gen.yaml

Reply:
OK. I withdraw this approach.